service/src/main/java/uk/gov/hmcts/reform/fpl/service/MessageJudgeService.java

```diff
@@ -298,65 +44,7 @@ private String buildMessageHistory(String message, String history, String sender
         return join("\n \n", history, formattedLatestMessage);
     }
 
-    private DynamicList rebuildJudicialMessageDynamicList(CaseData caseData, UUID selectedC2Id) {
-        return caseData.buildJudicialMessageDynamicList(selectedC2Id);
-    }
-
-    private boolean hasC2s(CaseData caseData) {
-        return caseData.getC2DocumentBundle() != null;
-    }
-
-    private boolean hasAdditionalApplications(CaseData caseData) {
-        return ObjectUtils.isNotEmpty(getApplications(caseData));
-    }
-
-    private boolean hasPlacementApplications(CaseData caseData) {
-        return ObjectUtils.isNotEmpty(caseData.getPlacementEventData().getPlacements());
-    }
-
-    private boolean hasSelectedAdditionalApplication(CaseData caseData) {
-        return (hasAdditionalApplications(caseData) || hasC2s(caseData) || hasPlacementApplications(caseData))
-            && caseData.getMessageJudgeEventData().getAdditionalApplicationsDynamicList() != null;
-    }
-
-    private boolean hasJudicialMessages(CaseData caseData) {
-        return !caseData.getJudicialMessages().isEmpty();
-    }
-
-    private boolean hasMatchingReplyEmaiAddress(JudicialMessage judicialMessageReply) {
-        return isNotEmpty(judicialMessageReply.getReplyFrom())
-            && judicialMessageReply.getReplyFrom().equals(judicialMessageReply.getReplyTo());
-    }
-
-    private boolean isReplyingToMessage(JudicialMessage judicialMessageReply) {
-        return judicialMessageReply != null && YES.getValue().equals(judicialMessageReply.getIsReplying());
-    }
-
-    private Map<String, Object> prePopulateSenderAndRecipient() {
-        Map<String, Object> data = new HashMap<>();
-
-        if (userService.hasUserRole(JUDICIARY)) {
-            data.put("judicialMessageMetaData", JudicialMessageMetaData.builder()
-                .sender(userService.getUserEmail())
-                .recipient(ctscEmailLookupConfiguration.getEmail()).build());
-        } else {
-            data.put("judicialMessageMetaData", JudicialMessageMetaData.builder()
-                .sender(ctscEmailLookupConfiguration.getEmail())
-                .recipient(EMPTY).build());
-        }
-
-        return data;
-    }
-
-    private String getReplyFrom() {
-        if (userService.hasUserRole(JUDICIARY)) {
-            return userService.getUserEmail();
-        } else {
-            return ctscEmailLookupConfiguration.getEmail();
-        }
-    }
-
-    private List<Element<SelectableItem>> getApplications(CaseData caseData) {
+    protected List<Element<SelectableItem>> getApplications(CaseData caseData) {
```

Comment:
Is this method only used in the SendNewMessageService?

Reply:
yes, the send message flow will query the applications if user are sending message related to an application. 
In reply message flow, the name and link of the related application is already stored in the JudicialMessage object.